Quick-Start [DA]/MVP/MVP+Makefile.md

```diff
@@ -1 +1,894 @@
+CREATE FUNCTION add(a integer, b integer) RETURNS integer
+    LANGUAGE SQL
+    IMMUTABLE
+    RETURNS NULL ON NULL INPUT
+    RETURN a + b;
+```
+
+### Step 4: Create the Makefile
+- Create a Makefile in the "extension" directory. 
+  - The Makefile is used to define the build process. It specifies the module name, extension name, and SQL files. The PG_CONFIG variable retrieves the path to the pg_config utility, which is used to get the necessary configuration for building the extension.
+
+```bash
+$ touch Makefile
+```
+- Open the Makefile and add the following content:
+
+```makefile
+EXTENSION = my_extension                 # Name of the extension
+DATA = my_extension--1.0.0.sql           # SQL file containing extension objects and functions
+
+PG_CONFIG  ?= pg_config                  # Path to the pg_config executable
+PGXS := $(shell $(PG_CONFIG) --pgxs)     # Get the PostgreSQL Extension Makefile from pg_config
+include $(PGXS)                          # Include the PostgreSQL Extension Makefile
+```
+
+<!--
+We could explain PGXS inshort to give the user more idea about the last 3 lines in the makefile.
+#### PGXS (PostgreSQL Extension Build Infrastructure) 
+It is a framework that helps simplify the process of building and installing extensions for PostgreSQL. It provides a standard way to compile and install extensions, making it easier to distribute and use them.
+
+To use PGXS:
+1. Create your extension's files, including source code and control files.
+2. Create a Makefile in your extension's directory.
+3. Inside the Makefile, include PGXS by adding a few lines of code.
+4. Define any extension-specific instructions in the Makefile.
+5. Use the `make` command to build and install your extension.
+6. Enable the extension in PostgreSQL using the `CREATE EXTENSION` SQL command.
+
+PGXS takes care of locating your PostgreSQL installation, determining compilation options, and ensuring the extension files are in the right place. It simplifies the process, making it easier for you to build and install your extension in different PostgreSQL environments.
+-->
+
+For more information on the Makefiles, you can refer to the [PostgreSQL documentation on makefile](https://www.postgresql.org/docs/15/extend-pgxs.html).
+
+#### Tip: 
+```makefile
+EXTENSION = my_extension
+DATA = my_extension--1.0.0.sql my_extension--1.1.0.sql
+```
+
+- By including multiple SQL script files in the `DATA` variable of the `Makefile`, the user can ensure that different versions of the extension can have different sets of objects or functionality. 
+- Each SQL script file represents a specific version of the extension and can contain the necessary SQL commands for the desired changes.
+- This allows the user to easily manage and control the installation of specific features or modifications in different versions of your extension.
+
+### Step 5: Build and Install the Extension
+- To build and install the extension, run the following commands:
```

Comment:
I think it's worth covering the need for`make` and PGXS, often installed via `-lib` packages in Yum, apt, and the like.

Reply:
I can cover the `make` explaination in the use of external files( C extensions ). Its wont be helpful for the user to see the whole explanation all here with out an example to back it.

I can add a whole section on using PGXN and add a link to this original QS. I was thinking that this thould be the main QS and all the other QS be linked to this central basic QS. Currently we have:
- QS on explaining Extensibility
- Procedural Language (plpython) [Need to add plpgsql]

- I will add a separate PGXN QS, which cover `pgxnclient`. It can link to the PGXN docs too.
- External language is also left we will need to cover that as well (Python and C)

Reply:
I'm not referring to PGXN, but that the use of a `Makefile` requires `make` and the use of `PGXS` (not PGXN*N*) requires the Postgres development libraries.

Reply:
**Need for `make`:**
   - `make` is a build automation tool used to compile and build software projects, including PostgreSQL extensions.
   - It reads instructions from a file called "Makefile" to determine the build process, dependencies, and rules for compilation.
   - Without `make`, developers would need to manually manage complex compilation steps, leading to errors and inefficiencies. 

**How `make` Command Functions:**
   - The `make` command is executed in the extension's directory to initiate the build process defined in the Makefile.
   - It automatically compiles source files, links object files, and generates the shared object (`.so`) file for the extension.
   - By default, `make` uses the `all` target in the Makefile, but developers can specify other targets for specific tasks, such as cleaning or installing the extension.

**The Role of PGXS (PostgreSQL Extension Build Infrastructure):**
   - PGXS is a standardized build infrastructure provided by PostgreSQL for developing extensions independently of the core codebase.
   - It includes macros, rules, and utilities that streamline the extension development process and ensure compatibility with the installed PostgreSQL version.
   - PGXS simplifies the integration of extensions with the PostgreSQL installation, making it easier to distribute and manage extensions.

**Installation and Use of PGXS:**
   - The PGXS infrastructure is installed with the PostgreSQL development libraries and headers.
   - On Linux distributions, developers can install the necessary packages, such as `postgresql-server-dev-X`, via package managers like Yum or Apt.
   - PGXS provides the necessary environment and tools for extension development, ensuring compatibility with the installed PostgreSQL version. 

**Using a Makefile for Extension Building:**
   - When building a PostgreSQL extension, developers create a Makefile with instructions for compiling and linking extension code.
   - The Makefile uses PGXS macros to include PostgreSQL-specific build settings, libraries, and header files necessary for extension development.
   - The Makefile specifies the extension's shared object (`.so`) file, name, and version, ensuring proper integration with the PostgreSQL environment.

these are some of the points i could get from the previous reference documentation i made. @theory Could you guide me with points should i include?

Reply:
If you still have reference documentation, it'd be good to simply link to these bits, but honestly I think your readers will be best served by a list of instructions for how to install all the dependencies on RedHat-based Linux, Debian-based Linux, macOS, and Windows.